src/gitops/components/shared/ApplicationSetList.tsx

```diff
@@ -0,0 +1,443 @@
+            titleText="Unable to load data"
+            bodyText="There was an error retrieving applicationsets. Check your connection and reload the page."
+          />
+        </Td>
+      </Tr>
+    </Tbody>
+  );
+
+  let currentActiveState = null;
+  if (loadError) {
+    currentActiveState = DataViewState.error;
+  } else if (applicationSets.length === 0) {
+    currentActiveState = DataViewState.empty;
+  }
+
+  return (
+    <div>
+      {showTitle == undefined && (
+        <ListPageHeader title={t('plugin__gitops-plugin~ApplicationSets')}>
+          <ListPageCreate groupVersionKind={modelToRef(ApplicationSetModel)}>
+            Create ApplicationSet
+          </ListPageCreate>
+        </ListPageHeader>
+      )}
+      <ListPageBody>
+        {!hideNameLabelFilters && (
+          <ListPageFilter
+            data={data}
+            loaded={loaded}
+            rowFilters={filters}
+            onFilterChange={onFilterChange}
+          />
+        )}
+        <DataView activeState={currentActiveState}>
+          <DataViewTable
+            rows={rows}
+            columns={columnsDV}
+            bodyStates={loadError ? { error } : { empty }}
+          />
+        </DataView>
+      </ListPageBody>
+    </div>
+  );
+};
+
+const ApplicationSetActionsCell: React.FC<{ appSet: ApplicationSetKind }> = ({ appSet }) => {
+  const [actions] = useApplicationSetActionsProvider(appSet);
+  return (
+    <div style={{ textAlign: 'right' }}>
+      <ActionsDropdown
+        actions={actions}
+        id="gitops-applicationset-actions"
+        isKebabToggle={true}
+      />
+    </div>
+  );
+};
+
+const useApplicationSetRowsDV = (applicationSetsList, namespace, applications, appsLoaded): DataViewTr[] => {
+  const rows: DataViewTr[] = [];
+  applicationSetsList.forEach((appSet: ApplicationSetKind, index: number) => {
+    rows.push([
+      {
+        cell: (
+          <div>
+            <ResourceLink
+              groupVersionKind={modelToGroupVersionKind(ApplicationSetModel)}
+              name={appSet.metadata.name}
+              namespace={appSet.metadata.namespace}
+              inline={true}
+            />
+          </div>
+        ),
+        id: appSet.metadata?.name,
+        dataLabel: 'Name',
+      },
+      ...(!namespace
+        ? [
+            {
+              cell: <ResourceLink kind="Namespace" name={appSet.metadata.namespace} />,
+              id: appSet.metadata.namespace,
+              dataLabel: 'Namespace',
+            },
+          ]
+        : []),
+      {
+        id: getAppSetStatus(appSet),
+        cell: (
+          <div>
+            <ApplicationSetStatusFragment status={getAppSetStatus(appSet)} />
+          </div>
+        ),
+      },
+      {
+        id: getGeneratedAppsCount(appSet, applications, appsLoaded).toString(),
+        cell: getGeneratedAppsCount(appSet, applications, appsLoaded).toString(),
+      },
+      {
+        id: getAppSetGeneratorCount(appSet).toString(),
```

Comment:
Should these individual columns have a unique id?  Look at others below this, which include the value of index in them